internal/host/host.go

```diff
@@ -1282,36 +1282,72 @@ func (m *Manager) selectRole(ctx context.Context, h *models.Host, db *gorm.DB) (
 		}
 	}
 
+	if expectedMasterCount < common.MinMasterHostsNeededForInstallationInHaMode &&
+		expectedMasterCount >= common.MinMasterHostsNeededForInstallationInHaArbiterMode {
+		arbiterCountNotIncludingHost := countNumberOfHostsInRoleNotIncludingHost(h, cluster, models.HostRoleArbiter)
+
+		if arbiterCountNotIncludingHost == 0 {
```

Comment:
Why not include this in the previous condition? We don't seem to be using this value anywhere else.